hoodie-utilities/src/main/java/com/uber/hoodie/utilities/HoodieDataImporter.java

```diff
@@ -0,0 +1,236 @@
+        try {
+            // Get commit time.
+            String commitTime = HoodieActiveTimeline.getNewCommitTime();
+
+            //Get schema.
+            FileSystem fs = FSUtils.getFs();
+            String schemaStr = getSchema(fs);
+
+            // Initialize target hoodie table.
+            Properties properties = new Properties();
+            properties.put(HoodieTableConfig.HOODIE_TABLE_NAME_PROP_NAME, cfg.tableName);
+            properties.put(HoodieTableConfig.HOODIE_TABLE_TYPE_PROP_NAME, cfg.tableType);
+
+            HoodieTableMetaClient.initializePathAsHoodieDataset(fs, cfg.targetPath, properties);
+
+            Path folderPath = new Path(cfg.srcPath);
+            RemoteIterator<LocatedFileStatus> locatedFileStatusRemoteIterator = fs
+                .listFiles(folderPath, true);
+            SQLContext sqlContext = new SQLContext(jsc);
+            JavaRDD<WriteStatus> writeStatuses = jsc.emptyRDD();
+            HoodieWriteClient client = createHoodieClient(jsc, cfg.targetPath, schemaStr, cfg.parallelism);
+
+            Job job = Job.getInstance(jsc.hadoopConfiguration());
+            ParquetInputFormat.setReadSupportClass(job, (AvroReadSupport.class));
+
+            JavaRDD<GenericRecord> genericRecords = jsc
+                .newAPIHadoopFile(cfg.srcPath, ParquetInputFormat.class, Void.class,
+                    GenericRecord.class, job.getConfiguration()).values();
+            JavaRDD<HoodieRecord<HoodieAvroPayload>> hoodieRecords = genericRecords.map(
+                new Function<GenericRecord, HoodieRecord<HoodieAvroPayload>>() {
+                    @Override
+                    public HoodieRecord<HoodieAvroPayload> call(GenericRecord genericRecord)
+                        throws Exception {
+
+                        Object partitionField = genericRecord.get(cfg.partitionKey);
+                        if (partitionField == null) {
+                            throw new HoodieDataImporterException(
+                                "partition key is missing. :" + cfg.partitionKey);
+                        }
+                        Object rowField = genericRecord.get(cfg.rowKey);
+                        if (rowField == null) {
+                            throw new HoodieDataImporterException(
+                                "row field is missing. :" + cfg.rowKey);
+                        }
+                        long ts = (long) ((Double) partitionField* 1000l);
+                        String partitionPath = HoodieWriteClient.PARTITION_FORMATTER
+                            .format(new Date(ts));
+                        return new HoodieRecord<HoodieAvroPayload>(
+                            new HoodieKey((String)rowField, partitionPath),
+                            new HoodieAvroPayload(Optional.of(genericRecord)));
+                    }
+                }
+            );
+            JavaRDD<WriteStatus> writeResponse = client.bulkInsert(hoodieRecords, commitTime);
```

Comment:
@prazanna Just to make it clear; You mean in case of an error we should cleanup the "targetPath" and retry. Right? Only problem I see in this is that most of the time when it will fail it will fail for valid reasons. So should we allow to retry or just cleanup the directory and exit?